project-generated-code-backend/Backend/Repository/DatabaseSql/RoomRenovationDatabaseSql.cs

```diff
@@ -0,0 +1,61 @@
+﻿using HealthClinicBackend.Backend.Model.Schedule;
+using HealthClinicBackend.Backend.Repository.Generic;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace HealthClinicBackend.Backend.Repository.DatabaseSql
+{
+    public class RoomRenovationDatabaseSql : GenericDatabaseSql<RoomRenovation>, IRoomRenovationRepository
+    {
+        public RoomRenovationDatabaseSql() : base()
+        {
+        }
+
+        public RoomRenovationDatabaseSql(HealthCareSystemDbContext dbContext) : base(dbContext)
+        {
+        }
+
+        public override List<RoomRenovation> GetAll()
+        {
+            return DbContext.RoomRenovations.ToList();
+        }
+
+        public override RoomRenovation GetById(string id)
+        {
+            return DbContext.RoomRenovations.Find(id);
+        }
+
+
+        public override void Save(RoomRenovation newEntity)
+        {
+            DbContext.RoomRenovations.Add(newEntity);
+            DbContext.SaveChanges();
+        }
+
+        public override void Update(RoomRenovation updateEntity)
+        {
+            DbContext.RoomRenovations.Update(updateEntity);
+            DbContext.SaveChanges();
+        }
+
+        public override void Delete(string id)
+        {
+            var roomRenovation = GetById(id);
+            if (roomRenovation == null) return;
+            DbContext.RoomRenovations.Remove(roomRenovation);
+            DbContext.SaveChanges();
+        }
+
+        public override RoomRenovation GetBySerialNumber(string serialNumber)
+        {
+            return GetAll().Where(f => f.SerialNumber.Equals(serialNumber)).ToList()[0];
```

Comment:
you can use DbContext.RoomRenovation.Find because serial number is the primary key